src/main/java/com/blamejared/crafttweaker/impl/commands/crafttweaker/ConflictCommands.java

```diff
@@ -0,0 +1,261 @@
+import com.blamejared.crafttweaker.impl.helper.ItemStackHelper;
+import com.mojang.brigadier.builder.LiteralArgumentBuilder;
+import net.minecraft.command.CommandSource;
+import net.minecraft.command.Commands;
+import net.minecraft.entity.player.PlayerEntity;
+import net.minecraft.item.ItemStack;
+import net.minecraft.item.crafting.IRecipe;
+import net.minecraft.item.crafting.IRecipeType;
+import net.minecraft.item.crafting.RecipeManager;
+import net.minecraft.util.ResourceLocation;
+import net.minecraft.util.concurrent.ThreadTaskExecutor;
+import net.minecraft.util.text.StringTextComponent;
+import net.minecraft.util.text.TextFormatting;
+import net.minecraftforge.fml.LogicalSide;
+import net.minecraftforge.fml.LogicalSidedProvider;
+import net.minecraftforge.fml.common.thread.EffectiveSide;
+import org.apache.logging.log4j.util.TriConsumer;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.NoSuchElementException;
+import java.util.PrimitiveIterator;
+import java.util.Spliterator;
+import java.util.Spliterators;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.function.Function;
+import java.util.function.Predicate;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+import java.util.stream.StreamSupport;
+
+public final class ConflictCommands {
+    
+    private static final class DescriptiveFilter implements Predicate<Map.Entry<ResourceLocation, IRecipe<?>>> {
+        
+        private final Predicate<IRecipe<?>> delegate;
+        private final String description;
+        
+        DescriptiveFilter(final Predicate<IRecipe<?>> delegate, final String description) {
+            
+            this.delegate = delegate;
+            this.description = description;
+        }
+        
+        @Override
+        public boolean test(final Map.Entry<ResourceLocation, IRecipe<?>> recipe) {
+        
+            return this.delegate.test(recipe.getValue());
+        }
+    
+        String description() {
+            
+            return this.description;
+        }
+    }
+    
+    private static final class RecipeIterator implements PrimitiveIterator.OfLong {
+        
+        private final int size;
+        
+        private int currentLeft;
+        private int currentRight;
+        private boolean kill;
+        
+        RecipeIterator(final int size) {
+            
+            this.size = size;
+            this.currentLeft = 0;
+            this.currentRight = this.currentLeft + 1;
+            this.kill = false;
+        }
+        
+        @Override
+        public long nextLong() {
+        
+            if (this.kill) throw new NoSuchElementException();
+            
+            final long current = make(this.currentLeft, this.currentRight);
+            if (++this.currentRight >= this.size) {
```

Comment:
I'm never too sure about pre/post increments in "larger" expressions.
Should we extract the side effect?
```suggestion
            this.currentRight++;
            if (this.currentRight >= this.size) {
```